src/core/io/platform/f4xx/ADCf4xx.cpp

```diff
@@ -84,109 +117,224 @@ void ADCf4xx::initADC(uint8_t num_channels) {
         break;
     case Pin::PC_2:
-        adcChannel.Channel = ADC_CHANNEL_12;
+        channelStruct = {.adc1 = 1, .adc2 = 1, .adc3 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_12)};
         break;
     case Pin::PC_3:
-        adcChannel.Channel = ADC_CHANNEL_13;
+        channelStruct = {.adc1 = 1, .adc2 = 1, .adc3 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_13)};
         break;
     case Pin::PC_4:
-        adcChannel.Channel = ADC_CHANNEL_14;
+        channelStruct = {.adc1 = 1, .adc2 = 1, .adc3 = 0, .channel = static_cast<uint8_t>(ADC_CHANNEL_14)};
         break;
     case Pin::PC_5:
-        adcChannel.Channel = ADC_CHANNEL_15;
+        channelStruct = {.adc1 = 1, .adc2 = 1, .adc3 = 0, .channel = static_cast<uint8_t>(ADC_CHANNEL_15)};
         break;
     default:
+        channelStruct = {}; // sets all values to 0
+        log::LOGGER.log(log::Logger::LogLevel::ERROR, "INVALID PIN 0x%x!!", pin);
         break; // Should never get here
     }
 
+    // This checks if the pin being used supports the ADC being used
+    if (checkSupport(adcPeriph, channelStruct)) {
+        adcChannel.Channel = channelStruct.channel;
+    } else {
+        log::LOGGER.log(log::Logger::LogLevel::ERROR, "ADC %d DOES NOT SUPPORT PIN 0x%x!!", (adcNum + 1), pin);
+    }
+
     // Subtract 1 because rank starts at 1
-    channels[rank - 1] = pin;
+    adcState.channels[rank - 1] = pin;
 
     adcChannel.Rank         = rank;
-    adcChannel.SamplingTime = ADC_SAMPLETIME_480CYCLES;
+    adcChannel.SamplingTime = ADC_SAMPLETIME_3CYCLES;
     adcChannel.Offset       = 0;
-    adcChannel.Offset       = 0x000;
 
-    HAL_ADC_ConfigChannel(&halADC, &adcChannel);
+    HAL_ADC_ConfigChannel(&adcState.halADC, &adcChannel);
+}
+
+bool ADCf4xx::checkSupport(ADCPeriph periph, Channel_Support channelStruct) {
+    // In c++, non-zero values (like 1) are true, and 0 is false, so no comparison is needed.
+    switch (periph) {
+    case ADCPeriph::ONE:
+        return channelStruct.adc1;
+    case ADCPeriph::TWO:
+        return channelStruct.adc2;
+    case ADCPeriph::THREE:
+        return channelStruct.adc3;
+    }
+}
+
+uint8_t ADCf4xx::getADCNum(ADCPeriph periph) {
+    switch (periph) {
+    case ADCPeriph::ONE:
+        return ADC1_SLOT;
+    case ADCPeriph::TWO:
+        return ADC2_SLOT;
+    case ADCPeriph::THREE:
+        return ADC3_SLOT;
+    }
+}
+
+/*
+ * This method initializes timer 8 with a Trigger Output of "Update Trigger", with a timer frequency of 1kHz.
+ * The timer does not specifically tell the exact ADC to do a conversion, it just sends a general Update Trigger every
+ * cycle. The ADC's are listening for this Update Trigger from timer 8, which is set in the ADC initialization.
+ */
+void ADCf4xx::initTimer() {
+    TIM_ClockConfigTypeDef sClockSourceConfig = {0};
+    TIM_MasterConfigTypeDef sMasterConfig     = {0};
+
+    ADCf4xx::htim8.Instance               = TIM8;
```

Comment:
because the timer is static and without the ADCf4xx:: in front of it it wont build cause it tries to call it as the instances htim, but that doesnt exist. The other places where I added the ADCf4xx are for the same reason.

Reply:
Ok, sounds good